rllib/connectors/connector_v2.py

```diff
@@ -149,8 +127,232 @@ def __call__(
+                            "match the single-agent contents in the given list of "
+                            "(multi-agent) episodes!"
+                        )
+                    d = data[key][list_indices[key]]
+                    list_indices[key] += 1
+                    yield sa_episode, d
+                else:
+                    yield sa_episode
+
+    @staticmethod
+    def add_batch_item(
+        batch: Dict[str, Any],
+        column: str,
+        item_to_add: Any,
+        single_agent_episode: Optional[SingleAgentEpisode] = None,
+    ) -> None:
+        """Adds a data item under `column` to the given `batch`.
+
+        If `single_agent_episode` is provided and it contains agent ID and module ID
+        information, will store the item in a list under a `([agent_id],[module_id])`
+        key within `column`. In all other cases, will store the item in a list directly
+        under `column`.
+
+        .. testcode::
+
+            from ray.rllib.connectors.connector_v2 import ConnectorV2
+            from ray.rllib.utils.test_utils import check
+
+            batch = {}
+            ConnectorV2.add_batch_item(batch, "test_col", 5)
+
+            check(batch, {"test_col": [5]})
+
+            sa_episode = SingleAgentEpisode(agent_id="ag1", module_id="module_10")
+            ConnectorV2.add_batch_item(batch, "test_col_2", -10, sa_episode)
+
+            check(batch, {
+                "test_col": [5],
+                "test_col_2": {
+                    "ag1:module_10": [-10],
+                },
+            })
+
+        Args:
+            batch: The batch to store `item_to_add` in.
+            column: The column name (str) within the `batch` to store `item_to_add`
+                under.
+            item_to_add: The data item to store in the batch.
+            single_agent_episode: An optional SingleAgentEpisode. If provided and its
+                agent_id and module_id properties are not None, will create a further
+                sub dictionary under `column`, mapping from `([agent_id],[module_id])`
+                (str) to a list of data items. Otherwise, will store `item_to_add`
+                in a list directly under `column`.
+        """
+        sub_key = None
+        if (
+            single_agent_episode
+            and single_agent_episode.agent_id is not None
+            and single_agent_episode.module_id is not None
+        ):
+            sub_key = (single_agent_episode.agent_id, single_agent_episode.module_id)
+
+        if column not in batch:
+            batch[column] = ([] if sub_key is None else {sub_key: []})
+        if sub_key:
+            batch[column][sub_key].append(item_to_add)
+        else:
+            batch[column].append(item_to_add)
+
+    @staticmethod
+    def switch_batch_from_agent_ids_to_module_ids(batch):#, episodes):
+        """Flips the mapping in the given batch from Agent ID based to Module ID based.
```

Comment:
Super nice, like this aggregation of data for module training becomes super easy. Especially when agent_id -> module_id is n -> m

Reply:
Yes, we'll have to poke around this a bit more in a follow-up PR with more (multi-agent) examples, but yes, this is the idea! :)